pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -130,17 +128,46 @@ func (d *DependenciesDistributor) NeedLeaderElection() bool {
 
 func (d *DependenciesDistributor) discoverResources(period time.Duration) {
 	wait.Until(func() {
-		for _, gvr := range supportedTypes {
-			if d.InformerManager.IsHandlerExist(gvr, d.EventHandler) {
-				continue
+		newResources := lifted.GetDeletableResources(d.DiscoveryClientSet)
+		for gvr := range newResources {
+			if !d.InformerManager.IsHandlerExist(gvr, d.EventHandler) && !d.gvrDisabled(gvr) {
+				klog.Infof("Setup informer for %s", gvr.String())
+				d.InformerManager.ForResource(gvr, d.EventHandler)
 			}
-			klog.Infof("Setup informer for %s", gvr.String())
-			d.InformerManager.ForResource(gvr, d.EventHandler)
 		}
 		d.InformerManager.Start()
 	}, period, d.stopCh)
 }
 
+// gvrDisabled returns whether GroupVersionResource is disabled.
+func (d *DependenciesDistributor) gvrDisabled(gvr schema.GroupVersionResource) bool {
```

Comment:
I think this [function gvrDisabled](https://github.com/karmada-io/karmada/blob/master/pkg/detector/detector.go#L174) should be reused. We can extract it into a public function.

Reply:
Updated.